examples/lfric/full_example/testkern_w0_mod.f90

```diff
@@ -33,20 +28,39 @@
 ! OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 ! OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ! -----------------------------------------------------------------------------
-! Modified R. W. Ford, STFC Daresbury Lab
-!
-module log_mod
-  implicit none
-  integer, public, parameter :: LOG_LEVEL_ALWAYS  = 100000
-  integer, public, parameter :: LOG_LEVEL_ERROR   = 200
-  integer, public, parameter :: LOG_LEVEL_WARNING = 150
-  integer, public, parameter :: LOG_LEVEL_INFO    = 100
-  integer, public, parameter :: LOG_LEVEL_DEBUG   =  50
-  integer, public, parameter :: LOG_LEVEL_TRACE   =   0
-  contains
-  subroutine log_event(message, level)
-    implicit none
-    character (*), intent( in ) :: message
-    integer,       intent( in ) :: level
-  end subroutine log_event
-end module log_mod
+! Author R. Ford STFC Daresbury Lab
+
+module testkern_w0_mod
+  use argument_mod
+  use kernel_mod
+  use fs_continuity_mod, only: W0
+
+  use constants_mod
+  type, extends(kernel_type) :: testkern_w0_type
+     type(arg_type), dimension(2) :: meta_args =  &
+          (/ arg_type(gh_field,gh_inc, w0),       &
+             arg_type(gh_field,gh_read,w0)        &
+           /)
+     integer :: iterates_over = cells
+   contains
+     procedure, nopass :: code => testkern_w0_code
+  end type testkern_w0_type
+contains
+
+  subroutine testkern_w0_code(nlayers, fld1, fld2, ndf_w0, undf_w0, map_w0)
+    integer :: nlayers
+    real(kind=r_def), dimension(:), intent(inout) :: fld1
+    real(kind=r_def), dimension(:), intent(in)    :: fld2
+    integer :: ndf_w0, undf_w0
+    integer, dimension(:) :: map_w0
+
+    integer :: i, k
+    do i=1, ndf_w0
+      do k=0, nlayers-1
```

Comment:
`k` loop over layers is the outermost one in LFRic kernels.

Reply:
OK ... really? That looks SO wrong, but ok, changed. (considering that the k-loop is vectorised).